ast/edit/testcase.go

```diff
@@ -1,60 +1,241 @@
 package edit
 
 import (
+	"context"
 	"fmt"
+	"path/filepath"
+	"strings"
 
 	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/dependencies/filesystem"
+	"github.com/influxdata/flux/parser"
 )
 
-/* TestCaseTransform is a macro that transforms ast that uses
-   the testcase statement into multiple independent packages with
-   a single flux script that is the test.
-
-   For instance, the following flux:
-
-import "testing"
-myVar = 4
-testcase addition { test.assertEqual(want: 2 + 2, got: myVar }
-
-	...would be transformed into the equivalent script...
-
-import "testing"
-myVar = 4
-test.assertEqual(want: 2 + 2, got: myVar
-*/
-func TestcaseTransform(pkg *ast.Package) ([]*ast.Package, error) {
+// TestCaseTransform will transform an *ast.Package into a set of *ast.Package values
+// that represent each testcase defined within the original package.
+//
+// A testcase is defined with the testcase statement such as below.
+//
+//     import "testing/assert"
+//     myVar = 4
+//     testcase addition {
+//         assert.equal(want: 2 + 2, got: myVar)
+//     }
+//
+// This gets transformed into a package that looks like this:
+//
+//     import "testing/assert"
+//     myVar = 4
+//     assert.equal(want: 2 + 2, got: myVar)
+//
+// It is allowed to include options within the testcase block as they will be extracted
+// to the top level.
+//
+// In addition to this syntax, testcase blocks may also include imports to other test files.
+// These imports will transform the imported testcase in a slightly different way.
+// If a testcase is included by import, it will be translated such as:
+//
+//     import "testing/assert"
+//     math_test = () => {
+//         myVar = 4
+//         test_addition = () => {
+//             assert.equal(want: 2 + 2, got: myVar)
+//             return {}
+//         }
+//         return {myVar, test_addition}
+//     }()
```

Comment:
Is this the generated AST being shown? If so what is the input AST that generates this output?

Reply:
It's the generated AST in the case that the above test were imported.